front/components/assistant/Sharing.tsx

```diff
@@ -0,0 +1,331 @@
+      title: "Moving to Company Assistants",
+      text: "Moving the assistant to Company Assistants will make the assistant editable only by Admins and Builders and add it to every member's “My Assistants” list.",
+      confirmText: "Move to Company",
+      variant: "primary",
+    },
+  },
+  private: {
+    label: "Personal Assistant",
+    color: "sky",
+    icon: LockIcon,
+    text: "Only I can view and edit.",
+    confirmationModalData: {
+      title: "Moving to Personal Assistants",
+      text: `Moving the assistant to your Personal Assistants will make the assistant unaccessible to other members of the workspace.`,
+      confirmText: "Move to Personal",
+      variant: "primaryWarning",
+      showUsage: true,
+    },
+  },
+  global: {
+    label: "Default Assistant",
+    color: "slate",
+    icon: DustIcon,
+    text: "Default assistant provided by Dust.",
+    confirmationModalData: null,
+  },
+};
+
+export function SharingSection({
+  owner,
+  agentConfigurationId,
+  initialScope,
+  newScope,
+  setNewScope,
+}: {
+  owner: WorkspaceType;
+  agentConfigurationId: string | null;
+  initialScope: Exclude<AgentConfigurationScope, "global">;
+  newScope: Exclude<AgentConfigurationScope, "global">;
+  setNewScope: (scope: Exclude<AgentConfigurationScope, "global">) => void;
+}) {
+  const agentUsage = useAgentUsage({
+    workspaceId: owner.sId,
+    agentConfigurationId,
+  });
+  const { agentConfiguration } = useAgentConfiguration({
+    workspaceId: owner.sId,
+    agentConfigurationId,
+  });
+  const assistantName = agentConfiguration?.name;
+
+  const usageText = assistantName
+    ? assistantUsageMessage({
+        assistantName,
+        usage: agentUsage.agentUsage,
+        isLoading: agentUsage.isAgentUsageLoading,
+        isError: agentUsage.isAgentUsageError,
+      })
+    : "";
+
+  return (
+    <div className="flex flex-col gap-3">
+      <div className="text-lg font-bold text-element-900">Sharing</div>
+      <SharingDropdown
+        owner={owner}
+        agentConfigurationId={agentConfigurationId}
+        initialScope={initialScope}
+        newScope={newScope}
+        setNewScope={setNewScope}
+      />
+      <div className="text-sm text-element-700">
+        <div>{SCOPE_INFO[newScope].text}</div>
+        {agentUsage &&
+        agentUsage.agentUsage?.userCount &&
+        agentUsage.agentUsage.userCount > 1
+          ? usageText
+          : null}
+      </div>
+    </div>
+  );
+}
+
+/*
+ * Note: Non-builders cannot change to/from company assistant
+ */
+export function SharingDropdown({
+  owner,
+  agentConfigurationId,
+  disabled,
+  initialScope,
+  newScope,
+  setNewScope,
+}: {
+  owner: WorkspaceType;
+  agentConfigurationId: string | null;
+  disabled?: boolean;
+  initialScope: AgentConfigurationScope;
+  newScope: AgentConfigurationScope;
+  setNewScope: (scope: Exclude<AgentConfigurationScope, "global">) => void;
+}) {
+  const { agentConfiguration } = useAgentConfiguration({
```

Comment:
Can we avoid an extra `useAgentConfiguration` here?

Reply:
wdym? we need it below for name + inmylist, + if it's used by a parent component it's cached so not costly right?

Reply:
or would you pass an agentconfigobject directly?